clients/roscpp/src/libros/service_server_link.cpp

```diff
@@ -371,7 +372,10 @@ bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& re
     }
   }
 
-  info->call_finished_ = true;
+  if (!info->call_finished_)
+  {
+    info->call_finished_ = true;
+  }
```

Comment:
What is this change for?

Reply:
If `call_finished_` is changed in https://github.com/ros/ros_comm/blob/089dd64a6c2d7769d2d8d3b10808eec621a3acf3/clients/roscpp/src/libros/service_server_link.cpp#L253, we don't need to set it with true again.

Reply:
But if the value is `true` already why is it a problem to just assign it to `true` again?

Reply:
There exist some failure cases that `callFinished` not to be called, please refer to https://github.com/ros/ros_comm/pull/2074#discussion_r504347435

Reply:
@dirk-thomas i was the one who originally mentioned about this change. you are correct. at the moment, i was thinking that this is expected to be true, so check if it is false with debug print and make it true then proceed. but as you mentioned, we can just assign it to true again. @iuhilnehc-ynos what do you think?

Reply:
> what do you think?

Okay, I'll update it.